moveit_ros/planning/planning_scene_monitor/src/current_state_monitor.cpp

```diff
@@ -263,7 +263,24 @@ bool planning_scene_monitor::CurrentStateMonitor::haveCompleteState(const ros::D
   return result;
 }
 
-bool planning_scene_monitor::CurrentStateMonitor::waitForCurrentState(double wait_time) const
+bool planning_scene_monitor::CurrentStateMonitor::waitForCurrentState(const ros::Time t, double wait_time) const
+{
+  ros::WallTime start = ros::WallTime::now();
+  ros::WallDuration elapsed;
+  ros::WallDuration timeout(wait_time);
+
+  boost::mutex::scoped_lock lock(state_update_lock_);
+  while (current_state_time_ < t)
+  {
+    state_update_condition_.wait_for(lock, boost::chrono::nanoseconds((timeout-elapsed).toNSec()));
+    elapsed = ros::WallTime::now() - start;
+    if (elapsed > timeout)
+      return false;
+  }
+  return true;
+}
+
+bool planning_scene_monitor::CurrentStateMonitor::waitForCompleteState(double wait_time) const
```

Comment:
So in this PR you took an old function ``waitForCurrentState`` and renamed it ``waitForCompleteState``? Then created a new function ``waitForCurrentState`` that collides with the old function's name? This sounds incredibly convoluted - shouldn't you retain the old function's name here and create a new function with new functionality under a new name??